src/helix.idea.vim

```diff
@@ -0,0 +1,189 @@
+nmap <Space>b <Action>(RecentFiles)
+xmap <Space>b <Action>(RecentFiles)
+nmap <Space>g <Action>(RecentChangedFiles)
+xmap <Space>g <Action>(RecentChangedFiles)
+
+nmap <Space>G <Action>(Debug)
+xmap <Space>G <Action>(Debug)
+
+nmap <Space>k <Action>(QuickJavaDoc)
+xmap <Space>k <Action>(QuickJavaDoc)
+
+nmap <Space>s <Action>(FileStructurePopup)
+xmap <Space>s <Action>(FileStructurePopup)
+nmap <Space>S <Action>(GotoSymbol)
+xmap <Space>S <Action>(GotoSymbol)
+
+nmap <Space>d <Action>(ActivateProblemsViewToolWindow)
+xmap <Space>d <Action>(ActivateProblemsViewToolWindow)
+nmap <Space>D <Action>(ActivateProblemsViewToolWindow)
+xmap <Space>D <Action>(ActivateProblemsViewToolWindow)
+
+nmap <Space>c <Action>(CommentByLineComment)
+xmap <Space>c <Action>(CommentByLineComment)
+nmap <Space>C <Action>(CommentByBlockComment)
+xmap <Space>C <Action>(CommentByBlockComment)
+
+nmap <Space>y <Action>(EditorCopy)
+xmap <Space>y <Action>(EditorCopy)
+nmap <Space>R <Action>(EditorPaste)
+xmap <Space>R <Action>(EditorPaste)
+
+nmap <Space>r <Action>(RenameElement)
+xmap <Space>r <Action>(RenameElement)
+
+nmap <Space>a <Action>(ShowIntentionActions)
+xmap <Space>a <Action>(ShowIntentionActions)
+
+nmap <Space>j <Action>(RecentLocation)
+xmap <Space>j <Action>(RecentLocation)
+
+nmap <Space>/ <Action>(FindInPath)
+xmap <Space>/ <Action>(FindInPath)
+nmap <Space>? <Action>(GotoAction)
+xmap <Space>? <Action>(GotoAction)
+
+nmap <Space>z <Action>(ToggleDistractionFreeMode)
+
+" Window mode
+nmap <Space>ww <Action>(NextSplitter)
+xmap <Space>ww <Action>(NextSplitter)
+nmap <Space>wv <Action>(SplitVertically)
+xmap <Space>wv <Action>(SplitVertically)
+nmap <Space>ws <Action>(SplitHorizontally)
+xmap <Space>ws <Action>(SplitHorizontally)
+nmap <Space>wt <Action>(ChangeSplitOrientation)
+xmap <Space>wt <Action>(ChangeSplitOrientation)
+nmap <Space>wq <Action>(CloseEditor)
+xmap <Space>wq <Action>(CloseEditor)
+
+nmap <C-w>w <Action>(NextSplitter)
+xmap <C-w>w <Action>(NextSplitter)
+nmap <C-w>v <Action>(SplitVertically)
+xmap <C-w>v <Action>(SplitVertically)
+nmap <C-w>s <Action>(SplitHorizontally)
+xmap <C-w>s <Action>(SplitHorizontally)
+nmap <C-w>t <Action>(ChangeSplitOrientation)
+xmap <C-w>t <Action>(ChangeSplitOrientation)
+nmap <C-w>q <Action>(CloseEditor)
+xmap <C-w>q <Action>(CloseEditor)
+
+" Tab Shifter (https://plugins.jetbrains.com/plugin/7475-tab-shifter)
+nmap <Space>wk <Action>(TabShiftActions.MoveFocusUp)
+xmap <Space>wk <Action>(TabShiftActions.MoveFocusUp)
+nmap <Space>wh <Action>(TabShiftActions.MoveFocusLeft)
+xmap <Space>wh <Action>(TabShiftActions.MoveFocusLeft)
+nmap <Space>wj <Action>(TabShiftActions.MoveFocusDown)
+xmap <Space>wj <Action>(TabShiftActions.MoveFocusDown)
+nmap <Space>wl <Action>(TabShiftActions.MoveFocusRight)
+xmap <Space>wl <Action>(TabShiftActions.MoveFocusRight)
+
+"
+" Non-standard workarounds
+"
+
+" repeat "vn" to select next search match
+xnoremap vn :action SelectNextOccurrence<CR>
```

Comment:
What is the difference with `n`?

Reply:
If I search eg. "noremap", on normal mode it selects the next occurrence (as it should), but if I change to visual mode, `n` does exactly the same - it does not retain the previous selection. This allowed to add a cursor to next instance.

I did some more testing and this would behave exactly like in Helix in select mode:

```vim
xmap n <Action>(SelectNextOccurrence)
```

But it then breaks how `n` works in normal mode. To make things even more frustrating, I just noticed that the IDE default `F3` works in normal mode exactly as on Helix. It shows up as `FindNext` when I enable "IdeaVim: Track action ID's". I then thought this would work:

```vim
xmap n <Action>(SelectNextOccurrence)
nmap n <Action>(FindNext)
```

...but no. When it's called like this, `FindNext` activates visual mode, even though `F3` does not. :unamused:

I'll remove this mapping. `helix.vim` default works as expected on normal mode and I'll personally using `<C-d>` to add a cursor to the next match out of old habit from the times I used VSCode I believe. Heck, I even do that in Helix:

```toml
[keys.normal]
'C-d' = ["search_selection", "extend_search_next"]

[keys.select]
'C-d' = ["search_selection", "extend_search_next"]
```

Reply:
Removed.